src/components/ProviderSelect.tsx

```diff
@@ -97,7 +100,15 @@ const ProviderSelect: FC<ProviderSelectProps> = ({
               const providerName = getName(provider);
 
               return (
-                <SelectOption key={providerName} value={provider}>
+                <SelectOption
+                  icon={
+                    providerTypeIcons(isDarkTheme)[
+                      provider?.spec?.type as keyof typeof providerTypeIcons
```

Comment:
nit: maybe the types for the util for providerTypeIcons should be updated so we don't have to cast here and in ProviderLinkCell?